share/jupyter/voila/templates/default/nbconvert_templates/voila.tpl

```diff
@@ -49,41 +49,52 @@ a.anchor-link {
 {{ spinner.html() }}
 
 <script>
-var voila_process = function(cell_index, cell_count) {
-  var el = document.getElementById("loading_text")
-  el.innerHTML = `Executing ${cell_index} of ${cell_count}`
-}
+  var voilaCellExecuteAfter = function(cell_index, cell_count) {};
+
+  var voilaCellExecuteBefore = function(cell_index, cell_count) {
+    var el = document.getElementById("loading_text");
+    el.innerHTML = `Executing ${cell_index} of ${cell_count}`;
+  };
+
+  // Backward compatibility
+  var voila_process = function(cell_index, cell_count) {};
 </script>
 
 <div id="rendered_cells" style="display: none">
 {%- endblock body_header -%}
 
 {%- block body_loop -%}
-{# from this point on, the kernel is started #}
-{%- with kernel_id = kernel_start() -%}
-  <script id="jupyter-config-data" type="application/json">
-  {
-      "baseUrl": "{{resources.base_url}}",
-      "kernelId": "{{kernel_id}}"
-  }
-  </script>
-  {% set cell_count = nb.cells|length %}
-  {#
-  Voila is using Jinja's Template.generate method to not render the whole template in one go.
-  The current implementation of Jinja will however not yield template snippets if we call a blocks' super()
-  Therefore it is important to have the cell loop in the template.
-  The issue for Jinja is: https://github.com/pallets/jinja/issues/1044
-  #}
-  {%- for cell in cell_generator(nb, kernel_id) -%}
-    {% set cellloop = loop %}
-    {%- block any_cell scoped -%}
-    <script>
-      voila_process({{ cellloop.index }}, {{ cell_count }})
+  {# from this point on, the kernel is started #}
+  {%- with kernel_id = kernel_start() -%}
+    <script id="jupyter-config-data" type="application/json">
+    {
+        "baseUrl": "{{resources.base_url}}",
+        "kernelId": "{{kernel_id}}"
+    }
     </script>
-      {{ super() }}
-    {%- endblock any_cell -%}
-  {%- endfor -%}
-{% endwith %}
+    {% set cell_count = nb.cells|length %}
+    {#
+    Voila is using Jinja's Template.generate method to not render the whole template in one go.
+    The current implementation of Jinja will however not yield template snippets if we call a blocks' super()
+    Therefore it is important to have the cell loop in the template.
+    The issue for Jinja is: https://github.com/pallets/jinja/issues/1044
+    #}
+    <script>voilaCellExecuteBefore(1, {{ cell_count }});</script>
+    {%- for cell in cell_generator(nb, kernel_id) -%}
+      {% set cellloop = loop %}
+      {%- block any_cell scoped -%}
+        <script>voila_process({{ cellloop.index }}, {{ cell_count }});</script>
+        {{ super() }}
+        <script>
+          voilaCellExecuteAfter({{ cellloop.index }}, {{ cell_count }});
+
+          if ({{ cellloop.index }} != {{ cell_count }}) {
```

Comment:
Pedantic mode: I think we should do the condition in jinja, so no code gets outputted, otherwise we get a:
```
if (3 != 3) {
```
In the output.